main/CommandHandler.cpp

```diff
@@ -2600,6 +2600,227 @@ int pref_getType(const uint8_t command[], uint8_t response[]) {
+    ret =  ESP_ERR_NO_MEM;
+    goto exit;
+  }
+  xSemaphoreGive(vhci_send_sem);
+
+  esp_bt_sleep_enable();
+
+  esp_vhci_host_register_callback(&vhciHostCb);
+
+exit:
+  response[2] = 1;          // number of parameters
+  response[3] = 1;          // length of first parameter
+  response[4] = ret;
+
+  return 6;
+}
+
+int ble_end(const uint8_t command[], uint8_t response[]) {
+  //[0]         CMD_START                   < 0xE0   >
+  //[1]         Command                     < 1 byte >
+
+  esp_bt_controller_disable();
+  esp_bt_controller_deinit();
+
+  if(buf_handle != NULL) {
+    vRingbufferDelete(buf_handle);
+  }
+
+  if (vhci_send_sem != NULL) {
+    /* Dummy take and give sema before deleting it */
+    xSemaphoreTake(vhci_send_sem, pdMS_TO_TICKS(2000));
+    xSemaphoreGive(vhci_send_sem);
+    vSemaphoreDelete(vhci_send_sem);
+    vhci_send_sem = NULL;
+  }
+
+  response[2] = 1;          // number of parameters
+  response[3] = 1;          // length of first parameter
+  response[4] = 1;
+
+  return 6;
+}
+
+int ble_available(const uint8_t command[], uint8_t response[]) {
+  //[0]         CMD_START                   < 0xE0   >
+  //[1]         Command                     < 1 byte >
+  uint16_t available = 0;
+  if(buf_handle != NULL) {
+    available = TO_HOST_BUF_SIZE - xRingbufferGetCurFreeSize(buf_handle);
+  }
+
+  response[2] = 1;          // number of parameters
+  response[3] = 2;          // length of first parameter
+  response[4] = (available >> 8)  & 0xff;
+  response[5] = (available >> 0)  & 0xff;
+
+  return 7;
+}
+
+int ble_peek(const uint8_t command[], uint8_t response[]) {
+  //[0]         CMD_START                   < 0xE0   >
+  //[1]         Command                     < 1 byte >
+  //[2]         N args                      < 1 byte >
+  //[3]         the number 2                < 1 byte >
+  //[4..5]      size                        < 2 byte >
+  // this could be useless xQueuePeek
+  uint8_t nargs = command[2];
+  // if nargs != 1 -> error
+  size_t res = 0;
+  // uint16_t size = ntohs(*((uint16_t *) &command[4]));
+  uint16_t size = *((uint16_t *) &command[4]);
+  uint8_t* received = nullptr;
+
+  if(size > TO_HOST_BUF_SIZE - xRingbufferGetCurFreeSize(buf_handle)) {
+    size = 0;
+    goto exit;
+  }
+
+  received = (uint8_t*)xRingbufferReceiveUpTo(buf_handle, &res, pdMS_TO_TICKS(2000), size);
+
+  memcpy(&response[5], received, res);
+
+exit:
+  response[2] = 1;          // number of parameters
+  response[3] = (size >> 8)  & 0xff;
+  response[4] = (size >> 0)  & 0xff;
+
+  return 6 + res;
+}
+
+int ble_read(const uint8_t command[], uint8_t response[]) {
+  //[0]         CMD_START                   < 0xE0   >
+  //[1]         Command                     < 1 byte >
+  //[2]         N args                      < 1 byte >
+  //[3]         the number 2                < 1 byte >
+  //[4..5]      size                        < 2 byte >
+  uint8_t nargs = command[2];
```

Comment:
@andreagilardoni I've found a couple of this unused nargs variables in this PR and another in the Preferences PR. Could you please add the check if needed, ore remove the unused variable?